ui/components/app/confirmations/alerts/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -19,14 +25,44 @@ import { useI18nContext } from '../../../../../hooks/useI18nContext';
 import useAlerts from '../../../../../hooks/useAlerts';
 import { AlertModal } from '../alert-modal';
 import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import useConfirmationAlertActions from '../../../../../pages/confirmations/hooks/useConfirmationAlertActions';
+
+type AlertActionHandlerContextType = {
+  processAction: (actionKey: string) => void;
+};
+
+const AlertActionHandlerContext = createContext<
+  AlertActionHandlerContextType | undefined
+>(undefined);
+
+export const useAlertActionHandler = () => {
+  const context = useContext(AlertActionHandlerContext);
+  if (!context) {
+    throw new Error(
+      'useAlertActionHandler must be used within an AlertActionHandlerProvider',
+    );
+  }
+  return context;
+};
+
+export const AlertActionHandlerProvider: React.FC<{
```

Comment:
This is perfect, but is all of this better placed in its own file in `ui/hooks` for decoupling?

That would be the context, the hook, and the provider?

Reply:
Great point, moved it to `UseAertActionHandler.ts` and added suitable tests.